xbmc/video/dialogs/GUIDialogTeletext.cpp

```diff
@@ -79,6 +79,29 @@ bool CGUIDialogTeletext::OnMessage(CGUIMessage& message)
   return CGUIDialog::OnMessage(message);
 }
 
+void CGUIDialogTeletext::DoProcess(unsigned int currentTime, CDirtyRegionList &dirtyregions)
+{
+  if (m_pTxtTexture)
+  {
+    m_TextDecoder.RenderPage();
+
+    unsigned char* textureBuffer = (unsigned char*)m_TextDecoder.GetTextureBuffer();
+    if (!m_bClose && m_TextDecoder.NeedRendering() && textureBuffer)
+    {
+      m_pTxtTexture->Update(m_TextDecoder.GetWidth(), m_TextDecoder.GetHeight(), m_TextDecoder.GetWidth()*4, XB_FMT_A8R8G8B8, textureBuffer, false);
+      m_TextDecoder.RenderingDone();
+    }
+
+    if (m_bClose || m_TextDecoder.NeedRendering() || teletextFadeAmount < 100)
```

Comment:
Good question. m_bClose is part of the Teletext dialog and initiates the fade out procedure. It worked when adding this check. I am currently investigating into your remark regarding RenderPage. It seems to be a bit more complicated.